src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/PurposePersistentBehavior.scala

```diff
@@ -321,7 +321,12 @@ object PurposePersistentBehavior {
       }
 
       val updatedVersion =
-        version.copy(state = newState, updatedAt = Some(timestamp), firstActivationAt = firstActivationAt)
+        version.copy(
+          state = newState,
+          updatedAt = Some(timestamp),
+          firstActivationAt = firstActivationAt,
+          suspendedAt = stateChangeDetails.timestamp
```

Comment:
this could be calculated like the `firstActivationAt` value.

Note: keeping mandatory the `timestamp` in `StateChangeDetails` allows you to remove line 315, and use the received timestamp.